app-misc/elasticsearch/elasticsearch-8.2.0.ebuild

```diff
@@ -0,0 +1,89 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit systemd tmpfiles
+
+DESCRIPTION="Free and Open, Distributed, RESTful Search Engine"
+HOMEPAGE="https://www.elastic.co/elasticsearch/"
+SRC_URI="https://artifacts.elastic.co/downloads/${PN}/${P}-linux-x86_64.tar.gz"
+LICENSE="Apache-2.0 BSD-2 Elastic-2.0 LGPL-3 MIT public-domain"
+SLOT="0"
+KEYWORDS="~amd64"
+
+RDEPEND="acct-group/elasticsearch
+	acct-user/elasticsearch
+	sys-libs/zlib
+	>=virtual/jdk-17:*"
```

Comment:
I'm still not convinced about this. Without inheriting a `java*.eclass` depending on `virtual/jdk-17:*`  seems useless. In this case its only purpose would be pulling-in `dev-java/openjdk`.
Have you tried with directly depending on `dev-java/openjdk:17`?

Reply:
That'll work, but (IMO) `virtual/jdk-17` is functionally equivalent to `|| ( dev-java/openjdk:17 dev-java/openjdk-bin:17 )` and is no harder to understand. It also comes with the `gentoo-vm` use flag which would make that `or` more unwieldy.

I'm happy to make the change if you'd prefer it that way though (or if I've missed why this is the preferred syntax).

Reply:
I suggest to state the dependencies more defensively, as we do with e.g. Python: we don't know if this will run on Java 19, hence this should be `virtual/jre:17` (or `virtual/jdk:17`, but I still don't see any evidence that this requires a JDK, but unfortunately I am unable to test this, as we don't have a Java 17 JRE in ::gentoo currently).